src/exampleTypes/ResultTable/index.js

```diff
@@ -88,10 +88,47 @@ let ResultTable = ({
     criteria,
   }
 
+  let columnGroupsHeight = _.flow(
+    _.map('fieldGroup.length'),
+    _.max
+  )(visibleFields)
+
+  let columnGroups = _.reduce(
+    (columnGroups, { fieldGroup, HeaderCell }) => {
+      for (let i = 0; i < columnGroupsHeight; i++) {
+        let groupRow = columnGroups[i] || (columnGroups[i] = [])
+        let groupName = _.getOr('', i, fieldGroup)
+        let lastGroup = _.last(groupRow)
+        if (_.get('groupName', lastGroup) === groupName) {
+          lastGroup.colspan++
+          lastGroup.HeaderCell = HeaderCell
+        } else groupRow.push({ groupName, colspan: 1, HeaderCell })
+      }
+      return columnGroups
+    },
+    [],
+    visibleFields
+  )
+
   return (
     <>
       <Table data-path={node.path}>
         <Thead>
+          {F.mapIndexed(
+            (columnGroupRow, i) => (
+              <Tr key={i}>
+                {F.mapIndexed(
+                  ({ groupName, colspan, HeaderCell = Th }, j) => (
+                    <HeaderCell key={j} colSpan={colspan}>
```

Comment:
do we need `i` an `j` to ensure a unique `key`?

Reply:
`key` only needs to be unique within this array

Reply:
Yeah your right, I missed the `Tr` the parent `F.mapIndexed` is nesting these in.